M2/Macaulay2/m2/matrix1.m2

```diff
@@ -400,7 +400,12 @@ RingElement ** RingElement := (r,s) -> matrix {{r}} ** matrix {{s}}
 -- Matrix#AfterPrint = Matrix#AfterNoPrint = f -> (class f, " ", new MapExpression from {target f,source f})
 Matrix#AfterPrint = Matrix#AfterNoPrint = f -> (
     class f,
-    if isFreeModule target f and isFreeModule source f then  (" ", new MapExpression from {target f,source f})
+    (
+	-- print source/target if both are free modules and/or have
+	-- been assigned to global variables
+	modulenames := select({target f, source f}, M -> (
+		isFreeModule M or hasAttribute(M, ReverseDictionary)));
+	if #modulenames == 2 then (" ", new MapExpression from modulenames))
```

Comment:
In response to your worry about documentation, how about revert `expression(Module)` to what it was, but keep printing the global name for matrix afterprints:
```suggestion
        (tar, src) := apply((target f, source f),
            M -> if isFreeModule M then M
            else if hasAttribute(M, ReverseDictionary)
            then getAttribute(M, ReverseDictionary));
        if tar =!= null and src =!= null
        then (" ", new MapExpression from expression \ {tar, src}))
```

Reply:
Actually, it would be nice to also use the global symbol in resolutions. Maybe in addition to the above, `net(Module)` should call `describe` instead, and we make sure that in printing of resolutions we call `expression M` not just `net M`. Would that solve the documentation issue?

Reply:
It's not just about the documentation. I tried using M2 with these changes for a couple of days because I really liked having non-free modules in resolutions being displayed simply by their symbol, but I actually **really** dislike having to type `describe M` every time I want to see the structure of a module outside of a resolution. This is not really the case for rings because typically users don't have a ton of rings around anyway.

Also, adding useful afterprints for matrices and vectors is a small addition, but changing a major behavior like printing of modules probably requires a larger conversation.

Reply:
> Actually, it would be nice to also use the global symbol in resolutions.

Do we ever have resolutions w/ non-free modules?

Reply:
I do:
```m2
i17 : R = kk[x,y];

i18 : chainComplex apply(1..2, i -> truncate_3 koszul_i vars R)

o18 = image | y3 xy2 x2y x3 | <-- image {1} | y2 xy x2 0  0  0  | <-- image {2} | y x |
                                        {1} | 0  0  0  y2 xy x2 |      
      0                                                               2
                                  1

o18 : ChainComplex
```

Reply:
But happy to add this as a local change if you want to limit the scope of this PR.